subsys/bluetooth/mesh/sol_pdu_rpl_srv.c

```diff
@@ -22,7 +22,6 @@ static void sol_rpl_status_rsp(const struct bt_mesh_model *mod,
 	BT_MESH_MODEL_BUF_DEFINE(buf, OP_SOL_PDU_RPL_ITEM_STATUS, 2 + (len < 2 ? 0 : 1));
 
 	bt_mesh_model_msg_init(&buf, OP_SOL_PDU_RPL_ITEM_STATUS);
```

Comment:
Why hasn't this been changed?

Reply:
I remember that I made the modification at the beginning, but the compiler reported an error. I guess it was caused by the conditional judgment statement, so there was no modification in the end.

Reply:
BTW：This is not a bugfix, it allows both ways to coexist, so even if a few are missed it doesn't seem to be a serious issue, that's what I think.

Reply:
> I remember that I made the modification at the beginning, but the compiler reported an error. I guess it was caused by the conditional judgment statement, so there was no modification in the end.

Similar conditional judgment statements were done here: https://github.com/zephyrproject-rtos/zephyr/pull/67662/files#diff-39c186341d5812c3a8edf4e2ba655115deb4b24100f60e438be7b97681d82e6cR117-R118
and here:  https://github.com/zephyrproject-rtos/zephyr/pull/67662/files#diff-39c186341d5812c3a8edf4e2ba655115deb4b24100f60e438be7b97681d82e6cR139-R140

What is the difference in this case?

> BTW：This is not a bugfix, it allows both ways to coexist, so even if a few are missed it doesn't seem to be a serious issue, that's what I think.

Yep, I understand it but when source code has the same approach it looks mature. Especially this one since Zephyr is open source project.

Reply:
Thank you for pointing it out

Reply:
Thank you for pointing it out

Reply:
Done.

Reply:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/26378606/2c25f462-b889-41c2-9553-11f025697264)

Reply:
@alxelax I will revert this, please see CI.

Reply:
/__w/zephyr/zephyr/subsys/bluetooth/mesh/sol_pdu_rpl_srv.c:22:9: error: variable-sized object may not be initialized